src/js/renderables/enemies/enemy.js

```diff
@@ -94,76 +84,56 @@ class Enemy extends me.Entity {
         }
     }
 
-    syncHitBox() {
-        this.hitbox.pos.x = this.pos.x;
-        this.hitbox.pos.y = this.pos.y;
-    }
-
     // Method to reduce the enemy's health when it takes damage
     takeDamage(damage) {
         this.health -= damage;
-
-        // If the health drops to 0 or below, call the die method
         if (this.health <= 0) {
             this.die();
         }
     }
 
     // Method to handle enemy death
     die() {
-        if(this.health <= 0) {
-
-        // Remove the enemy from the game world when it dies
-            this.onDestroy()
-
+        if(this.alive) {
```

Comment:
We don't need this.alive. Health should be enough to cause a death and isn't initialized anywhere.